confidential-data-hub/hub/src/storage/drivers/luks2.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2024 Intel
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+//! # LUKS2
+//!
+//! This module leverages cryptsetup to encrypt/decrypt a block device with luks2.
+//!
+//! It requires to install dependency `libcryptsetup-dev` for ubuntu.
+
+use std::path::Path;
+
+use libcryptsetup_rs::consts::flags::{CryptActivate, CryptDeactivate, CryptVolumeKey};
+use libcryptsetup_rs::consts::vals::EncryptionFormat;
+use libcryptsetup_rs::{CryptInit, CryptParamsLuks2, CryptParamsLuks2Ref};
+use zeroize::Zeroizing;
+
+/// Algorithm of the integrity hash
+const HMAC_SHA256: &str = "hmac-sha256";
+
+const SECTOR_SIZE: u32 = 4096;
+
+pub fn encrypt_device(
```

Comment:
It might be nice to do these functions as methods on a device struct. Not a requirement.